docs/rfcs/1012-parsec-auth.md

```diff
@@ -0,0 +1,432 @@
+
+  AUTH_MEDIUM_PUB_KEY --> |"Store in manifest"| ACCOUNT_MANIFEST
+
+  ACCOUNT_MANIFEST & ACCOUNT_MANIFEST_SYM_KEY --> E2{"Encrypt account manifest"} --> enc_MANIFEST["enc(ACCOUNT_MANIFEST)"]
+
+  AUTH_MEDIUM_PUB_KEY & ACCOUNT_MANIFEST_SYM_KEY --> E3{"Encrypt account manifest symmetric key"} --> enc_MANIFEST_KEY["enc(ACCOUNT_MANIFEST_SYM_KEY)"]
+
+  CSPRNG --> DEVICE_SYM_KEY
+
+  DEVICE_SYM_KEY & DEVICE --> E4{"Encrypt the device"} --> enc_DEVICE["enc(DEVICE)"]
+
+  DEVICE_SYM_KEY --> |"Store in manifest"| ACCOUNT_MANIFEST
+
+  subgraph Parsec-Auth server
+    AUTH_MEDIUM_HMAC_KEY
+    AUTH_MEDIUM_SYM_KEY_PARAMS
+    enc_PRIV_KEY
+    enc_MANIFEST
+    enc_MANIFEST_KEY
+    enc_DEVICE
+  end
+
+  E[Email] & P[Password] & AUTH_MEDIUM_SYM_KEY_PARAMS --> H3{PBKDF_2} --> AUTH_SYM_KEY[AUTH_MEDIUM_SYM_KEY]
+
+  AUTH_SYM_KEY & enc_PRIV_KEY --> D1{"Decrypt the private key"} --> PRIV_KEY[AUTH_MEDIUM_PRIV_KEY]
+
+  PRIV_KEY & enc_MANIFEST_KEY --> D2{"Decrypt the account manifest symmetric key"} --> MANIFEST_KEY[ACCOUNT_MANIFEST_SYM_KEY]
+
+  MANIFEST_KEY & enc_MANIFEST --> D3{"Decrypt the account manifest"} --> MANIFEST[ACCOUNT_MANIFEST]
+
+  MANIFEST --> DEV_KEY[DEVICE_SYM_KEY]
+
+  DEV_KEY & enc_DEVICE --> DEV[DEVICE]
+```
+
+## Alternatives Considered
+
+N/A
+
+## Operations
+
+That require managing a new service that will store the devices of the user.
+
+## Security/Privacy/Compliance
+
+Consideration should be taken about the method used to save the device on the remote server.
+The service would need to identify the user so we will need to have some information about the user (email mostly).
+
+### Consideration on PBKDF algorithms
+
+The `PBKDF_A` (used to derive a secret from the password) and `PBKDF_B` (used to derive the private key from the password) output should be different in order to avoid leaking the private key.
+
+To make the output different, we could:
+
+- Use different algorithms (e.g.: `argon2id`, `scrypt`)
+- Choose different salt (but they need to be deterministic from the information provided by the user only).
+- Use different parameters (but that may reduce the security)
+
+## Risks
+
+The devices are not stored securely to only allow the user to access them.
+In the case of using a password to login and to generate the asymmetric key pair,
+consideration should be taken to ensure that we obtain the key pair from the derivation of the password.
+
+In [Account creation](<#Account creation>), we use the password to generate 2 things:
+
+- An intermediate secret using `PBKDF_A` provided as is to the server.
+- The key pair using `PBKDF_B`.
+
+We have the section [Consideration of PBKDF algorithms](<#Consideration on PBKDF algorithms>) that state some of the requirement.
+
+## Remarks & open questions
+
+- The client application will need to communicate with the authentication service: should this be integrated into `libparsec`? Or it's the JS side that handle that?
```

Comment:
Since we already do client/server communication in `libparsec`, I think it's straightforward to continue doing that ^^

Reply:
I've noted you reply in the RFC